File: app/src/main/java/com/simpure/expires/data/entry/CommodityEntity.java
package com.simpure.expires.data.entry;

import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import com.simpure.expires.data.Date;
import com.simpure.expires.model.Commodity;

@Entity(tableName = "commodities")
public class CommodityEntity implements Commodity {
    @PrimaryKey
    private int id;
    private String name;
    private Long date;

    @Override
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Override
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public Long getDate() {
        return date;
    }

    public void setDate(Long date) {
        this.date = date;
    }


    public CommodityEntity() {
    }

    @Ignore
    public CommodityEntity(int id, String name, Long date) {
        this.id = id;
        this.name = name;
        this.date = date;
    }

    public CommodityEntity(Commodity commodity) {
        this.id = commodity.getId();
        this.name = commodity.getName();
        this.date = commodity.getDate();
    }
}

File: app/src/main/java/com/simpure/expires/data/entry/CommodityEntity.kt
package com.simpure.expires.data.entry

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import androidx.room.TypeConverters
import com.simpure.expires.data.CommodityType
import com.simpure.expires.data.Converters
import com.simpure.expires.data.InUse

import com.simpure.expires.model.CommodityDetail
import com.simpure.expires.model.Inventory
import com.simpure.expires.utilities.calcExpirationDate
import org.joda.time.DateTime
import org.joda.time.format.DateTimeFormat

@Entity(tableName = "commodities")
@TypeConverters(Converters::class)
class CommodityEntity constructor(
    @PrimaryKey
    override var id: Int,
    override var name: String,
    override var expirationDate: Long,
    override val unit: String = "",
    override val place: String = "",
    override val labels: List<String> = listOf(),
    override val barcode: String = "",
    override val inventories: List<Inventory> = listOf(),
    override val amount: Int = 0,
    override val inUse: InUse = InUse(),
    override val usedIdList: List<String> = listOf(),
    override val unboxingDuration: Long = 0L,
    override val disable: Boolean = false
) : CommodityDetail {

}
